tests/python/frontend/test_common.py

```diff
@@ -27,6 +32,203 @@ def test_key_is_not_present():
     assert not attrs.has_attr("b")
 
 
+def test_set_span():
```

Comment:
suggest to make this a TestCase class and promote the various e.g. `_verify_env_var_switch` functions to class methods e.g. `test_env_var_switch`, so they are separate test cases

Reply:
No problem!